app/assets/javascripts/oxalis/view/action-bar/save_button.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { connect } from "react-redux";
+import Model from "oxalis/model";
+import ButtonComponent from "oxalis/view/components/button_component";
+import type { OxalisState } from "oxalis/store";
+
+const SAVED_POLLING_INTERVAL = 100;
+
+class SaveButton extends React.PureComponent {
+  props: {
+    onClick: Function,
+  };
+
+  componentDidMount() {
+    // Polling can be removed once VolumeMode saving is reactive
+    this.savedPollingInterval = window.setInterval(this._forceUpdate, SAVED_POLLING_INTERVAL);
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.savedPollingInterval);
+  }
+
+  savedPollingInterval: number = 0;
+  _forceUpdate = () => {
+    this.forceUpdate();
+  };
+
+  getSaveButtonIcon() {
+    if (!Model.stateSaved()) {
+      return "hourglass";
+    } else {
+      return "check";
+    }
+  }
+
+  render() {
+    return (
+      <ButtonComponent
+        key="save-button"
+        type="primary"
+        onClick={this.props.onClick}
+        icon={this.getSaveButtonIcon()}
+      >
+        Save
+      </ButtonComponent>
+    );
+  }
+}
+
+function mapStateToProps(state: OxalisState) {
+  return {
+    save: state.save,
```

Comment:
It is not referenced directly. However, `Model.stateSaved()` interacts with the prop. In case of this component I use the `save` prop to initiate an update. `save` is changed whenever an action is pushed to the savequeue etc. Ideally, we would only need to listen to this prop without any of the `setTimeout` stuff... Could we leave it out? Perhaps, but then we wouldn'z get an instance update as soon as the user clicks on the button.